### DISCUSSION TITLE
[Question]RuntimeWarning: coroutine

### DISCUSSION THREAD

**Discussion by 7576457** [flet-dev/flet#2926] [Q&A] [answered]
### Question

Hello. The code I've attached is working fine. Everything is displaying correctly. However, I'm encountering an error in the console. What could be the problem?

### Code sample

```python
import asyncio

import flet as ft

class BotSheet(ft.BottomSheet):
    def __init__(self):
        super().__init__()
        self.is_scroll_controlled=True

    def did_mount(self):
        self.open = True
        self.page.run_task(self.build)

    def will_unmount(self):
        self.open = False

    async def build(self):
        self.content = ft.Container(height=300, bgcolor='red')
        self.update()
        await asyncio.sleep(1)
          

def main(page: ft.Page):
    
    page.add(
        ft.OutlinedButton('Click',            
        on_click=lambda e: page.show_bottom_sheet(BotSheet())
      )
    )

ft.app(main)
```

### Error message

```bash
RuntimeWarning: coroutine 'BotSheet.build' was never awaited innerCmds = ctrl._build_add_commands( RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment, marked as answer]
The `build` method shouldn't be made async.

Read this: https://python.plainenglish.io/whats-new-in-flet-0-21-0-ca482ab4520b

**Reply by 7576457** [reply]
Thank you for your article. With the update 0.21.0, UserControl is now considered deprecated. So, what does that make the 'def build()' method in the code I provided? I assumed it's now a method outside the UserControl context since I'm not inheriting from it anymore.

**Reply by FeodorFitsner** [reply]
Take a look at this, not yet published, article: https://github.com/flet-dev/website/blob/v2/docs/create/custom-controls.md

**Comment by GENRATECODE** [comment]
what did_mount also does not async
